sds_data_manager/stacks/processing_stack.py

```diff
@@ -0,0 +1,113 @@
+            Lambda directory
+        archive_bucket : s3.Bucket
+            S3 bucket
+        instrument_target : str
+            Target data product
+        instrument_sources : str
+            Data product sources
+        batch_security_group : ec2.SecurityGroup
+            Batch processor security group
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        self.batch_resources = FargateBatchResources(self,
+                                                     f"FargateBatchEnvironment-{sds_id}",
+                                                     sds_id,
+                                                     vpc=vpc,
+                                                     security_group=batch_security_group,
+                                                     processing_step_name=processing_step_name,
+                                                     archive_bucket=archive_bucket,)
+
+        self.instrument_lambda = InstrumentLambda(self, f"InstrumentLambda-{sds_id}",
+                                                  sds_id=sds_id,
+                                                  processing_step_name=processing_step_name,
+                                                  archive_bucket=archive_bucket,
+                                                  code_path=str(lambda_code_directory),
+                                                  instrument_target=instrument_target,
+                                                  instrument_sources=instrument_sources)
+
+        self.step_function = SdcStepFunction(self,
+                                             f"SdcStepFunction-{processing_step_name}",
+                                             processing_step_name=processing_step_name,
+                                             processing_system=self.instrument_lambda,
+                                             batch_resources=self.batch_resources,
+                                             instrument_target=instrument_target,
+                                             archive_bucket=archive_bucket)
+
+        # TODO: This will be a construct and also we will add to its capabilities.
+        rule = events.Rule(self, "rule",
+                           event_pattern=events.EventPattern(
+                               source=["aws.s3"],
+                               detail_type=["Object Created"],
+                               detail={
+                                   "bucket": {
+                                       "name": [archive_bucket.bucket_name]
+                                   },
+                                   "object": {
+                                       "key": [{
+                                           "prefix": f"{instrument_sources}"
+                                       }]}
+                               }
+                           ))
+
+        rule.add_target(event_targets.SfnStateMachine(self.step_function.state_machine))
```

Comment:
I think event bridge can pass inputs to the target. May be this is the place to pass in s3 object and bucket information. Then step function can pass that to lambda payload and to batch as well.

Reply:
Yep! That's the plan.